portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +25,35 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  public ArrayList<String> randomFacts;
+
+  @Override
+  public void init() {
+      randomFacts = new ArrayList<String>();
+      randomFacts.add("Hello Malachi");
+      randomFacts.add("Red");
+      randomFacts.add("Computer Science and Cognitive Science");
+  }
+    
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    init();
+    response.setContentType("application/json;");
+    String json = convertToJson(randomFacts);
+    response.getWriter().println(json);
+  }
+
+  public String convertToJson(ArrayList<String> randomFacts) {
```

Comment:
Since randomFacts is also the name of a class variable, it might help readability to name this parameter something else.

Although the method is referring to the randomFacts param rather than the randomFacts class var, when reading the method it could get confusing which is being referred to.

Reply:
I changed the name of randomFacts to comments, and made that variable private and final.

Reply:
Done.